### ISSUE TITLE
Windows Permission Error Errno 13

### ISSUE THREAD

**Issue by panchel** [fepegar/resector#3]
* resector version:  Bump version: 0.2.5 → 0.2.6
* Python version: 3.7.9
* Operating System: Windows x64

### Description
I have successfully downloaded resector, but when trying to run the resect t1... command, I keep getting "Permission Error:  [Errno 13] Permission denied: 'C:\\Users\\username\\AppData\\Local\\Temp\\tmpmy3cjap9.nii' ". The last NIFTI file name changes every time I try to run the command.

### What I Did

I have tried this on two separate PCs, used command prompt as an administrator, git bash as an administrator, changed the properties of the Appdata and Temp folders so all users have access, and it all has not worked.  From what I understand, most people using MacOS have no problem running resector. Is this a windows problem or is there anything else I can do?

```shell
resect /c/Users/username/Desktop/folder/mni_icbm152_t1_tal_nlin_asym_09c.nii /c/Users/username/Desktop/folder/17549_NeuroMorph_Parcellation.nii.gz /c/Users/username/Desktop/folder/outcome.nii.gz /c/Users/username/Desktop/folder/outcome_label.nii.gz
```

```python-traceback
Traceback (most recent call last):
  File "C:\Users\yingt\AppData\Local\Programs\Python\Python37\Scripts\resect-script.py", line 33, in <module>
    sys.exit(load_entry_point('resector', 'console_scripts', 'resect')())
  File "c:\users\yingt\appdata\local\programs\python\python37\lib\site-packages\click\core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "c:\users\yingt\appdata\local\programs\python\python37\lib\site-packages\click\core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "c:\users\yingt\appdata\local\programs\python\python37\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\users\yingt\appdata\local\programs\python\python37\lib\site-packages\click\core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "c:\users\yingt\resector\resector\cli\resect.py", line 59, in main
    parcellation_path,
  File "c:\users\yingt\resector\resector\cli\resect.py", line 126, in ensure_images
    parcellation_path, hemisphere)
  File "c:\users\yingt\resector\resector\parcellation.py", line 56, in get_gray_matter_mask
    mask = nib_to_sitk(array, parcellation_nii.affine) > 0
  File "c:\users\yingt\resector\resector\io.py", line 70, in nib_to_sitk
    nib.Nifti1Image(array, affine).to_filename(f.name)
  File "c:\users\yingt\appdata\local\programs\python\python37\lib\site-packages\nibabel\filebasedimages.py", line 333, in to_filename
    self.to_file_map()
  File "c:\users\yingt\appdata\local\programs\python\python37\lib\site-packages\nibabel\analyze.py", line 1030, in to_file_map
    hdrf = hdr_fh.get_prepare_fileobj(mode='wb')
  File "c:\users\yingt\appdata\local\programs\python\python37\lib\site-packages\nibabel\fileholders.py", line 70, in get_prepare_fileobj
    obj = ImageOpener(self.filename, *args, **kwargs)
  File "c:\users\yingt\appdata\local\programs\python\python37\lib\site-packages\nibabel\openers.py", line 113, in __init__
    self.fobj = opener(fileish, *args, **kwargs)
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\yingt\\AppData\\Local\\Temp\\tmp_3soszuh.nii'

```
Thank you

**Comment by fepegar** [comment]
Hi, @panchel. Thanks for reporting. I have never tried `resector` on Windows, but in theory it should work. It looks like you're having some OS issues. I'm not sure how to proceed. Can you try this?

```python
from pathlib import Path
from tempfile import NamedTemporaryFile

with NamedTemporaryFile(suffix='.txt') as f:
    print(f.name)
    path = Path(f.name)
    path.write_text('hello')
```

**Comment by panchel** [comment]
Thanks so much for looking into this problem. I tried running this code, but I'm getting the error: 
'from' is not recognized as an internal or external command,
operable program or batch file.

From what I understand, you are trying to create a temporary file and write in it.
I have previously been able to successfully create a file.txt into C:\\Users\\yingt\\AppData\\Local\\Temp, but I did not try writing in it. Please let me know if this helps or if there's anything else I can do.

**Comment by fepegar** [comment]
> Thanks so much for looking into this problem. I tried running this code, but I'm getting the error:
'from' is not recognized as an internal or external command,
operable program or batch file.

Can you try running it _inside_ Python? As opposed to the command line.

I have now access to a Windows machine so hopefully I will be able to solve this today.

**Comment by panchel** [comment]
Ah, I see, thank you! 

This is what I'm getting:
![temp file try](https://user-images.githubusercontent.com/86067371/122581608-3a92d200-d025-11eb-8d95-1fc6abaea347.JPG)

Interestingly, I tried running resector using anaconda with administrator access, and I'm still getting the permission error: 
![image](https://user-images.githubusercontent.com/86067371/122581904-8b0a2f80-d025-11eb-998b-fb3c85a53a92.png)

but it is also creating a bunch of tmp files in the Temp folder after I run the command, which I believe it was not doing before. 
![image](https://user-images.githubusercontent.com/86067371/122582052-b260fc80-d025-11eb-89df-96354ba57929.png)

Even after the resector stops running and shows the permission error, new tmp files are created into the Temp folder. Not sure why this is or if this is even related to resector, but it seems like this is a new thing happening.

**Comment by fepegar** [comment]
I have modified the code and hopefully I this is fixed in `v0.2.7`. I'm now getting another issue, which I think is unrelated. Please upgrade `resector` and let me know if you're still having issues.

**Comment by fepegar** [comment]
I'm now getting the error I posted at #6. You will probably get the same. I will try to look into that next week.
